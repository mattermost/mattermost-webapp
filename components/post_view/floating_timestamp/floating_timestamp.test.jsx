// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import FloatingTimestamp from './floating_timestamp.jsx';

describe('components/post_view/FloatingTimestamp', () => {
    const baseProps = {
        isScrolling: true,
        isMobile: true,
        createAt: 1234,
        toastPresent: true,
        isRhsPost: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<FloatingTimestamp {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.hasClass('toastAdjustment')).toBe(true);
    });
});
