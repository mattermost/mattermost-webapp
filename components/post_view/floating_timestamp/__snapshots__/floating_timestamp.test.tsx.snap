// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/FloatingTimestamp should match snapshot 1`] = `
<div
  className="post-list__timestamp scrolling toastAdjustment"
  data-testid="floatingTimestamp"
>
  <div>
    <span>
      <Connect(injectIntl(Timestamp))
        ranges={
          Array [
            Object {
              "display": <FormattedMessage
                defaultMessage="Today"
                id="date_separator.today"
              />,
              "equals": Array [
                "day",
                0,
              ],
            },
            Object {
              "display": <FormattedMessage
                defaultMessage="Yesterday"
                id="date_separator.yesterday"
              />,
              "equals": Array [
                "day",
                -1,
              ],
            },
          ]
        }
        useTime={false}
        value={1234}
      />
    </span>
  </div>
</div>
`;
